File: VB/CustomizeExportAction.Module.Web/Controllers/CustomizeExportControllerWeb.vb
Imports System
Imports System.ComponentModel
Imports System.Collections.Generic
Imports System.Diagnostics
Imports System.Text

Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Actions
Imports DevExpress.Persistent.Base

Imports DevExpress.ExpressApp.Web.SystemModule
'add a reference to the DevExpress.Web.Export assembly
Imports DevExpress.Web
Imports CustomizeExportAction.Module.Controllers

Namespace CustomizeExportAction.Module.Web.Controllers
   Partial Public Class CustomizeExportControllerWeb
       Inherits CustomizeExportController

      Protected Overrides Sub CustomExport(ByVal sender As Object, ByVal e As DevExpress.ExpressApp.SystemModule.CustomExportEventArgs)
         MyBase.CustomExport(sender, e)
         'Export only selected rows
         Dim exporter As ASPxGridViewExporter = TryCast(e.Printable, ASPxGridViewExporter)
         If exporter IsNot Nothing Then
            exporter.ExportSelectedRowsOnly = True
         End If
      End Sub
   End Class
End Namespace

File: CS/CustomizeExportAction.Module.Web/Controllers/CustomizeExportControllerWeb.cs
using System;
using System.ComponentModel;
using System.Collections.Generic;
using System.Diagnostics;
using System.Text;

using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Actions;
using DevExpress.Persistent.Base;

using DevExpress.ExpressApp.Web.SystemModule;
//add a reference to the DevExpress.Web.Export assembly
using DevExpress.Web;
using CustomizeExportAction.Module.Controllers;

namespace CustomizeExportAction.Module.Web.Controllers {
   public partial class CustomizeExportControllerWeb : CustomizeExportController {

      protected override void CustomExport(object sender, DevExpress.ExpressApp.SystemModule.CustomExportEventArgs e) {
         base.CustomExport(sender, e);
         //Export only selected rows
         ASPxGridViewExporter exporter = e.Printable as ASPxGridViewExporter;
         if (exporter != null) {
            exporter.ExportedRowType = GridViewExportedRowType.Selected;
         }
      }
   }
}
